### DISCUSSION TITLE
String is used as a range property but is not defined

### DISCUSSION THREAD

**Discussion by rljacobson** [terminusdb/terminusdb#296] [Q&A] [answered]
I sometimes get this error when using the schema creator to create a link property:
<img width="396" alt="Screen Shot 2021-04-10 at 11 29 30 AM" src="https://user-images.githubusercontent.com/1412598/114275498-bd2d5e80-99f0-11eb-8e5c-46bdd6ec2733.png">

I would expect the string datatype to automatically be defined. And why is the error about a string type when I am trying to create a link instance to another document, that is, no new strings are involved?

**Comment by Cheukting** [comment]
I think the reason is that if the range of a property is an xsd:sting, it should not be a link property? Link property is for object properties, i.e. an object (including documents) within the graph.

**Reply by rljacobson** [reply]
Yes, but my intention is to make a property that links to another document, so I am choosing Add Property --> Link Property.

I have prepared a multimedia presentation:

https://user-images.githubusercontent.com/1412598/114318476-1ec9f780-9adb-11eb-829d-75323dcce902.mp4

<img width="892" alt="Screen Shot 2021-04-11 at 3 27 52 PM" src="https://user-images.githubusercontent.com/1412598/114318308-99dede00-9ada-11eb-943f-f652b347dec2.png">

**Comment by spl** [comment, marked as answer]
This is no longer an issue in our latest release, v10.
